src/prefect/infrastructure/process.py

```diff
@@ -168,12 +168,21 @@ async def kill(self, infrastructure_pid: str, grace_seconds: int = 30):
                     f"Unable to kill process {pid!r}: The process was not found."
                 )
 
-            await anyio.sleep(grace_seconds)
+            for _ in range(grace_seconds):
+                await anyio.sleep(1)
```

Comment:
Should we do some fraction of the `grace_seconds` with a min of 1 or something? I'd like to avoid calling `kill` _too_ much as I'm not sure if it blocks the event loop.

Reply:
Sure, I'll see what I can do.